### PR TITLE
Add protobuf utilities for gRPC conversions

### PR DESCRIPTION
**Pull request by ZacharyEspiritu**
Part of #42. 

`FromGrpcStatusToStatus` was from `google-cloud-cpp` library, but can't use it directly since we had to redefine `StatusCode` ourselves.

### CODE REVIEW

**Comment by ZacharyEspiritu** [comment]
We might need to extend the timeout on Cloud Build for this (probably downloading OpenSSL + building gRPC is pushing it over the time limit).

**Review by tonyliaoss — APPROVED** [review]

File: src/backing/client/grpc_client/proto_util_test.cc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  EXPECT_EQ(expected, actual);
+}
+
+using ProtoDigestCase = google::cloud::kms::v1::Digest::DigestCase;
+
+// Mapping between `DigestCase` cases and their protobuf equivalents.
+struct CorrespondingDigestCase {
+  DigestCase expected;
+  ProtoDigestCase proto;
+} const kDigestMapping[]{
```
Comment (ajfeldma-google) [inline src/backing/client/grpc_client/proto_util_test.cc:128]: Nit: separate type definition from variable declaration as discussed previously

File: src/backing/client/grpc_client/proto_util_test.cc [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+using ProtoCryptoKeyVersion = google::cloud::kms::v1::CryptoKeyVersion;
+
+// Mapping between `CryptoKeyVersionAlgorithm` cases and their protobuf
+// equivalents.
+struct CorrespondingCryptoKeyVersionAlgorithm {
+  CryptoKeyVersionAlgorithm expected;
+  google::cloud::kms::v1::CryptoKeyVersion_CryptoKeyVersionAlgorithm proto;
+} kCryptoKeyVersionAlgorithmMapping[]{
```
Comment (ajfeldma-google) [inline src/backing/client/grpc_client/proto_util_test.cc:154]: Nit: separate type definition from variable declaration as discussed previously

File: src/backing/client/grpc_client/proto_util_test.cc [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  auto actual = MakeDigest(expected_digest_type, expected_digest_bytes);
+  EXPECT_EQ(FromProtoToDigestCase(actual.digest_case()), expected_digest_type);
+  EXPECT_THAT(GetDigestBytes(actual), StrEq(expected_digest_bytes));
+}
+
+// Mapping between `StatusCode` cases and their protobuf equivalents.
+struct CorrespondingStatusCode {
+  grpc::StatusCode proto;
+  StatusCode code;
+} kStatusCodeMapping[]{
```
Comment (ajfeldma-google) [inline src/backing/client/grpc_client/proto_util_test.cc:85]: Nit: separate type definition from variable declaration as discussed previously

**Review by ajfeldma-google — APPROVED** [review]

File: src/backing/client/grpc_client/proto_util_test.cc (hunk 1 above)
Comment (ZacharyEspiritu) [reply src/backing/client/grpc_client/proto_util_test.cc:128]: Handling in #77.

File: src/backing/client/grpc_client/proto_util_test.cc (hunk 2 above)
Comment (ZacharyEspiritu) [reply src/backing/client/grpc_client/proto_util_test.cc:154]: Handling in #77.

File: src/backing/client/grpc_client/proto_util_test.cc (hunk 3 above)
Comment (ZacharyEspiritu) [reply src/backing/client/grpc_client/proto_util_test.cc:85]: Handling in #77.

**Pull request merged**